llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -9926,6 +9926,42 @@ static SDValue getV4X86ShuffleImm8ForMask(ArrayRef<int> Mask, const SDLoc &DL,
   return DAG.getTargetConstant(getV4X86ShuffleImm(Mask), DL, MVT::i8);
 }
 
+// Canonicalize SHUFPD mask to improve chances of further folding.
```

Comment:
There's no further folding showing in test cases. Does it need a follow up patch?

Reply:
There are a number of test cases which now further fold to `VBLENDPD/S` (allowing commutation yay!) or `VUNPCK?PD` - the only further fold that leads to an actual instruction reduction so far is the `add_v4f64_024u_135u_reverse` test in vector-shuffle-256-v4.ll